### PR TITLE
CIV-7138 -  Mediation - Update content to Mediation Screen

### PR DESCRIPTION
**Pull request by neeta-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-7138

### Change description ###
This PR has code changes to update the content of Mediation screen

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java [hunk 1]
```diff
@@ -181,7 +185,61 @@ private CaseData setApplicantDefenceResponseDocFlag(CaseData caseData) {
     }
 
     private CallbackResponse setApplicantRouteFlags(CallbackParams callbackParams) {
-        CaseData updatedCaseData = setApplicantDefenceResponseDocFlag(setApplicant1ProceedFlagToYes(callbackParams.getCaseData()));
+        CaseData caseData = callbackParams.getCaseData();
+        CaseData updatedCaseData = setApplicantDefenceResponseDocFlag(setApplicant1ProceedFlagToYes(caseData));
+
+        /*((caseData.getRespondent1ClaimResponseTypeForSpec() == RespondentResponseTypeSpec.FULL_DEFENCE
```
Comment (kenneth-hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java:191]: Why this is comment?

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java [hunk 2]
```diff
@@ -181,7 +185,61 @@ private CaseData setApplicantDefenceResponseDocFlag(CaseData caseData) {
...
+
+    private void setMediationConditionFlag(CaseData updatedCaseData) {
+        if(isMediationRequired(updatedCaseData).equals(YES)) {
+            Set<DefendantResponseShowTag> showConditionFlags = updatedCaseData.getShowConditionFlags();
+            showConditionFlags.add(DefendantResponseShowTag.MEDIATION_2);
+            updatedCaseData.toBuilder().showConditionFlags(showConditionFlags);
+        }
+    }
+
+    private YesOrNo isMediationRequired(CaseData caseData) {
```
Comment (kenneth-hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java:214]: change to boolean as this method will not populate to ccd

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java [hunk 3]
```diff
@@ -181,7 +185,61 @@ private CaseData setApplicantDefenceResponseDocFlag(CaseData caseData) {
...
+        if(SpecJourneyConstantLRSpec.SMALL_CLAIM.equals(caseData.getResponseClaimTrack())
+            && caseData.getResponseClaimMediationSpecRequired().equals(YES))
+        {
+            switch (caseData.getRespondent1ClaimResponseTypeForSpec()) {
+                case FULL_DEFENCE:
+                    if(caseData.getApplicant1ProceedWithClaim().equals(YES)) {
```
Comment (kenneth-hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java:220]: YES.equals(caseData.getApplicant1ProceedWithClaim())

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java [hunk 4]
```diff
@@ -181,7 +185,61 @@ private CaseData setApplicantDefenceResponseDocFlag(CaseData caseData) {
...
+    private void setMediationConditionFlag(CaseData updatedCaseData) {
+        if(isMediationRequired(updatedCaseData).equals(YES)) {
+            Set<DefendantResponseShowTag> showConditionFlags = updatedCaseData.getShowConditionFlags();
+            showConditionFlags.add(DefendantResponseShowTag.MEDIATION_2);
+            updatedCaseData.toBuilder().showConditionFlags(showConditionFlags);
+        }
+    }
+
+    private YesOrNo isMediationRequired(CaseData caseData) {
+        if(SpecJourneyConstantLRSpec.SMALL_CLAIM.equals(caseData.getResponseClaimTrack())
```
[the remark this answers is not in the document]
Comment (neeta-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java]: done

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java (hunk 3 above)
Comment (neeta-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java]: done

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java (hunk 2 above)
Comment (neeta-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java]: done

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java (hunk 1 above)
Comment (neeta-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java]: code from other PR not yet merged to master

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/spec/show/DefendantResponseShowTag.java [hunk 5]
```diff
@@ -32,5 +32,6 @@ public enum DefendantResponseShowTag {
     WHY_1_DOES_NOT_PAY_IMMEDIATELY,
     WHY_2_DOES_NOT_PAY_IMMEDIATELY,
     REPAYMENT_PLAN_2,
-    MEDIATION
+    MEDIATION,
+    MEDIATION_2
```
[the remark this answers is not in the document]
Comment (neeta-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/spec/show/DefendantResponseShowTag.java]: done

**Review by kenneth-hmcts — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/enums/MediationDecision.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package uk.gov.hmcts.reform.civil.enums;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@Getter
+@RequiredArgsConstructor
+public enum MediationDecision {
```
Comment (dtortolaV1) [inline src/main/java/uk/gov/hmcts/reform/civil/enums/MediationDecision.java:8]: Why not to use existing enum YesOrNo?

File: src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/ClaimantMediationLip.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import uk.gov.hmcts.reform.civil.enums.MediationDecision;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class ClaimantMediationLip {
+
```
[the remark this answers is not in the document]
Comment (neeta-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/ClaimantMediationLip.java:14]: That field is used in defendant response journey. If we reuse it in claimant response journey that value will be erased. Hence I have added new field.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java [hunk 8]
```diff
@@ -187,7 +190,77 @@ private CaseData setApplicantDefenceResponseDocFlag(CaseData caseData) {
...
+            .data(updatedCaseData.toMap(objectMapper))
+            .build();
+    }
+
+    private void setMediationConditionFlag(CaseData updatedCaseData) {
+        DefendantResponseShowTag mediationFlag = setMediationRequired(updatedCaseData);
+        if (mediationFlag != null) {
+            Set<DefendantResponseShowTag> showConditionFlags = updatedCaseData.getShowConditionFlags();
+            showConditionFlags.add(mediationFlag);
+            updatedCaseData.toBuilder().showConditionFlags(showConditionFlags);
```
[the remark this answers is not in the document]
Comment (neeta-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java]: done

**Review by LeonardoPalmeiro — APPROVED** [review]

File: src/main/resources/caseIdForNotifyRpaOnCaseHandedOffline.txt [hunk 9]
```diff
@@ -1,10 +1,4 @@
-1671554345936878
-1671719628161672
-1673866074871359
-1674214750411548
-1675856602677905
-1676393623339520
-1656068208985525
+1671554345936878,1671719628161672,1673866074871359,1674214750411548,1675856602677905,16763936233395201656068208985525
```
Comment (kenneth-hmcts) [inline src/main/resources/caseIdForNotifyRpaOnCaseHandedOffline.txt:1]: Can you roll back the changes, I don't think it is related

**Review by dtortolaV1 — APPROVED** [review]

File: src/main/resources/caseIdForNotifyRpaOnCaseHandedOffline.txt (hunk 9 above)
Comment (neeta-hmcts) [reply src/main/resources/caseIdForNotifyRpaOnCaseHandedOffline.txt:1]: oh sure.

Comment (neeta-hmcts) [reply src/main/resources/caseIdForNotifyRpaOnCaseHandedOffline.txt]: done

**Review by dtortolaV1 — DISMISSED** [review]

**Review by kenneth-hmcts — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java [hunk 10]
```diff
@@ -179,19 +184,88 @@ private YesOrNo doesPartPaymentRejectedOrItsFullDefenceResponse(CaseData caseDat
...
+        DefendantResponseShowTag mediationFlag = setMediationRequired(caseData);
+        if (mediationFlag != null) {
+            Set<DefendantResponseShowTag> showConditionFlags = new HashSet<>(caseData.getShowConditionFlags());
+            showConditionFlags.add(mediationFlag);
+            updatedCaseData.showConditionFlags(showConditionFlags);
+        }
+    }
+
+    private DefendantResponseShowTag setMediationRequired(CaseData caseData) {
+        MultiPartyScenario multiPartyScenario = getMultiPartyScenario(caseData);
```
Comment (TatsyMaslova) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java:214]: A bit too much business logic in the handler with if and switch cases.. wonder whether we can refactor and also put it somewhere else for the handler to use. this way we can also unit test it separately

**Review by kenneth-hmcts — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java [hunk 11]
```diff
@@ -797,4 +800,42 @@ public boolean isRejectDefendantPaymentPlanNo() {
             || (YesOrNo.NO.equals(getApplicant1AcceptPartAdmitPaymentPlanSpec()))
             || !paymentPlan.contains(getDefenceAdmitPartPaymentTimeRouteRequired());
     }
+
+    @JsonIgnore
+    public boolean isDefendantHasNotPaidTheClaimant() {
+        return NO.equals(getApplicant1PartAdmitConfirmAmountPaidSpec());
```
Comment (TatsyMaslova) [inline src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java:806]: isDefendantNotPaid() or hasDefendantNotPaid() probably better?

Comment (neeta-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java:806]: done

File: src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java [hunk 12]
```diff
@@ -797,4 +800,42 @@ public boolean isRejectDefendantPaymentPlanNo() {
...
+    }
+
+    @JsonIgnore
+    public boolean isClaimantNotAgreeToSettleTheClaim() {
+        return NO.equals(getApplicant1PartAdmitIntentionToSettleClaimSpec());
```
[the remark this answers is not in the document]
Comment (neeta-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java:811]: done

File: src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java [hunk 13]
```diff
@@ -797,4 +800,42 @@ public boolean isRejectDefendantPaymentPlanNo() {
...
+        return NO.equals(getApplicant1AcceptAdmitAmountPaidSpec());
+    }
+
+    @JsonIgnore
+    public boolean isFullDefence() {
+        return YES.equals(getApplicant1ProceedWithClaim());
+    }
+
+    @JsonIgnore
+    public boolean isDefendantHasAcceptedMediation() {
```
[the remark this answers is not in the document]
Comment (neeta-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java]: defiantly better than mine :)

**Review by TatsyMaslova — APPROVED** [review]

**Review by kenneth-hmcts — DISMISSED** [review]

**Review by nagaraju-deekonda — APPROVED** [review]

**Review by laarmada — APPROVED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

**Review by rajakm — APPROVED** [review]

**Pull request merged**